File: board-room/src/main/java/net/stzups/board/room/server/websocket/protocol/server/messages/ServerMessageUpdateDocument.java
package net.stzups.board.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.board.data.objects.Document;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.board.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageUpdateDocument extends ServerMessage {
    private Document document;
    private boolean shared;

    public ServerMessageUpdateDocument(Document document) {
        this(document, false);
    }
    public ServerMessageUpdateDocument(Document document, boolean shared) {
        super(ServerMessageType.UPDATE_DOCUMENT);
        this.document = document;
        this.shared = shared;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        byteBuf.writeBoolean(shared);//takes up one byte
        byteBuf.writeLong(document.getId());
        writeString(document.getName(), byteBuf);
    }
}

File: scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageUpdateDocument.js
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";

export default class ServerMessageUpdateDocument extends ServerMessage {
    constructor(reader) {
        super(ServerMessageType.UPDATE_DOCUMENT);
        this.shared = reader.readUint8() ? 1 : 0;
        this.id = reader.readBigInt64();
        this.name = reader.readString8();
    }
}